SwiftAppTemplate/Screens/UserDetails/UserDetailsScreen.swift

```diff
@@ -0,0 +1,160 @@
+                .foregroundColor(.primaryNavyBlue)
+            Picker(getLocalString("GenderPlaceHolder"), selection: $selectedGender) {
+                ForEach(Gender.allCases) { gender in
+                    Text(getLocalString(gender.rawValue)).tag(gender)
+                }
+            }
+            .pickerStyle(SegmentedPickerStyle())
+        }
+    }
+    
+    var dobField: some View {
+        VStack(alignment: .leading){
+            
+            Text(getLocalString("DateOfBirth"))
+                .font(.notoSansMedium16)
+                .foregroundColor(.primaryNavyBlue)
+            CustomTextField(inputText: Binding.constant(formatDate(dateOfBirth)), placeholder: getLocalString("DateOfBirthPlaceHolder"), cornerRadius: 10, borderColor: .primaryNavyBlue)
+                .onTapGesture {
+                    self.showingDatePicker = true
+                }
+                .popover(isPresented: $showingDatePicker, attachmentAnchor: .point(.bottom), arrowEdge: .bottom) {
+                    VStack {
+                        DatePicker(
+                            getLocalString("DateOfBirthPlaceHolder"),
+                            selection: $dateOfBirth,
+                            displayedComponents: .date
+                        )
+                        .datePickerStyle(GraphicalDatePickerStyle())
+                        .labelsHidden()
+                        .frame(maxHeight: 400)
+                        
+                        Button(getLocalString("Done")) {
+                            self.showingDatePicker = false
+                        }
+                        .padding()
+                    }
+                    .padding()
+                }
+        }
+    }
+    
+    var countryField: some View {
+        VStack(alignment: .leading){
+            HStack {
+                Text(getLocalString("SelectCountry"))
+                    .font(.notoSansMedium16)
+                    .foregroundColor(.primaryNavyBlue)
+                Picker(getLocalString("SelectCountryPlaceHolder"), selection: $selectedCountry){
+                    ForEach(Constants.countriesOptions, id: \.self){
+                        country in
+                        Text(country)
+                    }
+                }
+                .pickerStyle(DefaultPickerStyle())
+                
+                Spacer()
+            }
+        }
+    }
+    
+    var languageField: some View {
+        VStack(alignment: .leading){
+            Text(getLocalString("SelectLanguage"))
+                .font(.notoSansMedium16)
+                .foregroundColor(.primaryNavyBlue)
+            CustomTextField(inputText: Binding.constant(selectedLanguage), placeholder: getLocalString("SelectLanguagePlaceHolder"), cornerRadius: 10, borderColor: .primaryNavyBlue)
+                .onTapGesture {
+                    openDeviceSettings()
+                }
+        }
+    }
+    
+    func hasEnteredAllDetails() -> Bool {
```

Comment:
Same for these functions as well, they can be extracted in a common util class perhaps?

Reply:
one difference UserDetailsScreen has no email field , but  Edit screen has email field